esrally/utils/io.py

```diff
@@ -274,24 +285,55 @@ def decompress(zip_name, target_directory):
     this function.
     """
     path_without_extension, extension = splitext(zip_name)
-    filename = basename(path_without_extension)
     if extension == ".zip":
         _do_decompress(target_directory, zipfile.ZipFile(zip_name))
     elif extension == ".bz2":
-        _do_decompress_manually(target_directory, filename, bz2.open(zip_name))
+        decompressor_args = ["pbzip2", "-d", "-k", "-m10000", "-c"]
```

Comment:
Does this mean we unconditionally allow `pbzip2` to use up to 10.000MB of memory? (`-m10000`). I wonder whether this might be a bit too aggressive for constrained environments (Docker containers?). On the other hand, having a lot of RAM available for the page cache later would be beneficial in any case so I'm a bit on the fence whether we should lower it. Did you by chance measure maximum memory usage of `pbzip2` while decompressing a corpus file?

Reply:
`-m` specifies the **max** memory usage; from the man page:

> Where # is max memory usage in 1MB steps (default 100 = 100MB)

So if an environment has less it will just whatever is available.

During the _compression+ with the same parameter (`-m10000`) on a machine with lots of ram (256GB) it reports:

```
$ pbzip2 -v -k -m10000 documents.json
Parallel BZIP2 v1.1.9     - by: Jeff Gilchrist [http://compression.ca]
[Apr. 13, 2014]               (uses libbzip2 by Julian Seward)
Major contributions: Yavor Nikolov <nikolov.javor+pbzip2@gmail.com>

         # CPUs: 12
 BWT Block Size: 900 KB
File Block Size: 900 KB
 Maximum Memory: 1410 MB
-------------------------------------------
         File #: 1 of 1
     Input Name: documents.json
    Output Name: documents.json.bz2

     Input Size: 79802445255 bytes
Compressing data...
```

whereas pmap shows:

```
                 Kbytes     RSS   Dirty 
---------------- ------- ------- -------
total kB          949904  114832  112048
```

i.e. seems to be using <1GB.

I haven't checked the usage during decompression yet.

Reply:
On the same machine during extraction `pmap -x` for the pbzip2 pid (`pbzip2 -d -k -m10000 -c /home/dliappis/.rally/benchmarks/data/nyc_taxis/documents.json.bz2`) reports:

`total kB          929564   77196   74656`

Reply:
Ok, then we should be ok. Thanks for double-checking.